clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -52,6 +54,19 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
     return {};
   }
 
+  /// Emits the address of the l-value, then loads and returns the result.
+  mlir::Value emitLoadOfLValue(const Expr *e) {
+    LValue lv = cgf.emitLValue(e);
+    // FIXME: add some akin to EmitLValueAlignmentAssumption(E, V);
+    return cgf.emitLoadOfLValue(lv, e->getExprLoc()).getScalarVal();
+  }
+
+  // l-values
+  mlir::Value VisitDeclRefExpr(DeclRefExpr *e) {
+    // TODO: Handle constant emission
```

Comment:
Should this/could this be a NYI diagnostic?

Reply:
The missing code here is a call to `tryEmitAsConstant()` and there's no simple way to test whether it will do anything. I'll put in a MissingFeature assert.